### PR TITLE
Update GitLab REST client library

### PR DESCRIPTION
**Pull request by AlexTugarev**
## Description
- [x] Execute gitlab tests manually

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at a8ec98a</samp>

Updated `@gitbeaker/node` dependency to fix GitLab pagination bug and moved `package.json` file to `components/server` directory to simplify workspace setup and dependency management.

</details>

## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - at-bump-gi5363ad8af0</li>
	<li><b>🔗 URL</b> - <a href="https://at-bump-gi5363ad8af0.preview.gitpod-dev.com/workspaces" target="_blank">at-bump-gi5363ad8af0.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - at-bump-gitlab-client-gha.16010</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by geropl** [comment]
@AlexTugarev I checked last time, and all versions of gitbeaker/node still pull the outdated packages.
IMO we need to switch to gitbeaker/rest (or however it was called).

**Comment by AlexTugarev** [comment]
@gitbeaker/rest is requiring to update to node 18.

**Comment by jeanp413** [comment]
What does it mean to `Execute gitlab tests manually`? connect with gitlab from dashboard and open some repo?

**Review by mustard-mh — APPROVED** [review]
Code LGTM, tried with project / workspace create, prebuild webhook worked well too 
image

**Comment by mustard-mh** [comment]
> What does it mean to `Execute gitlab tests manually`? connect with gitlab from dashboard and open some repo?

@jeanp413 Maybe is that in those `*.spec.ts` files related to Gitlab, most of them will skip if we don't provide a test token, so you can export this var if you want to `Execute gitlab tests manually`

```
skip(ifEnvVarNotSet("GITPOD_TEST_TOKEN_GITLAB")
```

**Comment by mustard-mh** [comment]
Once want to try with it, but test cases repo are owned by Alex, so I can't do test manually

File: components/server/src/gitlab/api.ts [hunk 1]
```diff
@@ -256,19 +255,19 @@ export namespace GitLab {
     export namespace Permissions {
         export function hasWriteAccess(repo: Project): boolean {
             if (repo.permissions.project_access) {
-                return repo.permissions.project_access.access_level >= 30;
+                return (repo.permissions.project_access as unknown as { access_level: number }).access_level >= 30;
```
Comment (AlexTugarev) [inline components/server/src/gitlab/api.ts:258]: Is no proper type export anymore? This looks like we can stop use typing at all. Let me double-check.

**Comment by AlexTugarev** [comment]
Just realized that branches are not retrievable. 

When trying to parse the following context, it fails fetching branch `wip`
`https://gitlab.com/AlexTugarev/gp-test/-/blob/wip/README.md`

File: components/server/src/gitlab/api.ts [hunk 2]
```diff
@@ -128,16 +115,16 @@ export namespace GitLab {
             return !!something && something.name === "GitLabApiError";
         }
         export function isNotFound(error: ApiError): boolean {
-            return !!error.httpError?.description.startsWith("404");
```
Comment (AlexTugarev) [inline components/server/src/gitlab/api.ts:131]: I've seen the `startsWith` of undefined before, and it was related to GitLab contexts. 🙏🏻 hopefully that's the cause

**Pull request merged**